rtrlib/lib/vector.c

```diff
@@ -0,0 +1,154 @@
+
+
+#include "vector.h"
+
+#include <math.h>
+
+int aspa_vector_create(struct aspa_vector **vector_pointer)
+{
+	const size_t DEFAULT_INITIAL_SIZE = 128;
+
+	// allocation the chunk of memory
+	struct aspas_record *data_field =
+		(struct aspas_record *)malloc(sizeof(struct aspas_record) * DEFAULT_INITIAL_SIZE);
+
+	// malloc failed so returning an error
+	if (data_field == NULL) {
+		return -1;
+	}
+
+	struct aspa_vector *vector = (struct aspa_vector *)malloc(sizeof(struct aspa_vector));
+
+	if (vector == NULL) {
+		return -1;
+	}
+
+	vector->capacity = DEFAULT_INITIAL_SIZE;
+	vector->size = 0;
+	vector->data = data_field;
+
+	*vector_pointer = vector;
+
+	return 0;
+}
+
+int aspa_vector_delete(struct aspa_vector *vector)
+{
+	// checking if one of the two pointers is potentially invalid
+	if (vector == NULL || vector->data == NULL) {
+		return -1;
+	}
```

Comment:
Doesn't that leak memory if the vector's data has been released but not the vector itself?

Reply:
yup, wonder if this needs to be thought of, but would certainly better if

Reply:
done